samcli/lib/providers/sam_stack_provider.py

```diff
@@ -170,15 +186,17 @@ def _convert_cfn_stack_resource(
         name: str,
         resource_properties: Dict,
         global_parameter_overrides: Optional[Dict] = None,
+        ignore_code_extraction_warnings: bool = False,
     ) -> Optional[Stack]:
         template_url = resource_properties.get("TemplateURL", "")
 
         if SamLocalStackProvider.is_remote_url(template_url):
-            LOG.warning(
-                "Nested stack '%s' has specified S3 location for Location which is unsupported. "
-                "Skipping resources inside this nested stack.",
-                name,
-            )
+            if not ignore_code_extraction_warnings:
```

Comment:
Feels weird that the method that's dealing with all the core logic has to leak its abstractions and ask callers to tell it whether there's something to warn about.

Reply:
I changed the implementation to not let stack provider aware of any warning. now its `get_stacks()` returns a tuple instead of a single list. The 2nd item of the tuple indicates the list of remote stacks detected, so that build context can warn accordingly.